### PR TITLE
Decouple UUID generation from SSDP discovery

### LINKED ISSUE #1078: UUID not generated if SSDP discovery is disabled
With discovery disabled, it should still be possible to use the SAT>IP server by providing the URL to `/desc.xml` manually. This currently only kind of works, because the XML is out of spec:

```
<UDN>uuid:</UDN>
```

This at least causes some older tvheadend versions to crash

### PR DESCRIPTION
**Pull request by Jalle19**
Fixes #1078, replaces #1119

### CODE REVIEW

**Review by catalinii — APPROVED** [review]

**Comment by Jalle19** [comment]
Needs runtime testing, seems to work but 🤷

File: src/minisatip.c [hunk 1]
```diff
@@ -1669,23 +1670,18 @@ int close_http(sockets *s) {
...
-    if (uuidi == 0) {
-        uuidi = 1;
-        get_mac_address(mac);
-        sprintf(uuid, "%s-%s", uuid1, mac);
-        // use IPv4 only as disc_host is multicast IPv4
-        fill_sockaddr(&ssdp_sa, opts.disc_host, 1900, 1);
-    }
+
+    // use IPv4 only as disc_host is multicast IPv4
+    fill_sockaddr(&ssdp_sa, opts.disc_host, 1900, 1);
```
Comment (Jalle19) [inline src/minisatip.c:1682]: I have a feeling that this is no longer correct (this should not be called all the time)

Comment (catalinii) [reply src/minisatip.c:1682]: The multicast address needs to be set somewhere in the ssdp_sa. Maybe is not the right place but it should be done, or am I missing something?

Comment (Jalle19) [reply src/minisatip.c:1681]: The fact that the variable `ssdp_sa` was global made me think there were some other uses for it, but it's apparently only used in this function so it should be fine to call `fill_sockaddr` many times.

**Comment by Jalle19** [comment]
Works as expected, UUID is generated correctly with `-G` applied.

**Pull request merged**